server/api/runtime_handlers/base.py

```diff
@@ -1059,7 +1072,9 @@ def _delete_pod_resources(
                         )
 
                 server.api.utils.singletons.k8s.get_k8s_helper().delete_pod(
-                    pod.metadata.name, namespace
+                    pod.metadata.name,
+                    namespace,
+                    grace_period_seconds=resource_deletion_grace_period,
```

Comment:
databricks jobs have a cancel task and if you force deletion here I believe it will not run. It needs to cancel the remote run so we might need to exclude databricks runtime here (see `termination_grace_period_seconds`)